source/runtime.cpp

```diff
@@ -3355,6 +3365,212 @@ void reshade::runtime::save_screenshot(const std::wstring &postfix)
 	}
 #endif
 }
+int reshade::runtime::execute_screenshot_postprocess(const std::filesystem::path &screenshot_path)
+{
+	int ec = 0;
+
+	if (_postprocess_application.empty())
+	{
+		ec = -1;
+		return ec;
+	}
+
+	if (_postprocess_application.extension() != L".exe" || !std::filesystem::is_regular_file(g_reshade_base_path / _postprocess_application))
+	{
+		LOG(ERROR) << "Executing the post-process program was blocked. Application path must be executable.";
+		ec = 1;
+		return ec;
+	}
+
+	std::string command_line;
+	command_line.append(1, '\"');
+	command_line.append((g_reshade_base_path / _postprocess_application).u8string());
+	command_line.append(1, '\"');
+
+	if (!_postprocess_arguments.empty())
+	{
+		command_line.append(1, ' ');
+
+		for (size_t cursor = 0; cursor < _postprocess_arguments.size();)
+		{
+			size_t brace_begin = _postprocess_arguments.find('<', cursor);
+			size_t brace_end = _postprocess_arguments.find('>', cursor + 1);
+
+			if (brace_begin == std::string::npos || brace_end == std::string::npos)
+			{
+				command_line.append(_postprocess_arguments.substr(cursor));
+				break;
+			}
+			else
+			{
+				command_line.append(_postprocess_arguments.substr(cursor, brace_begin - cursor));
+			}
+
+			std::string_view replacing = ((std::string_view)_postprocess_arguments).substr(brace_begin + 1, brace_end - (brace_begin + 1));
+			size_t colon_pos = replacing.find(":");
+
+			std::string name;
+			if (colon_pos == std::string::npos)
+				name = replacing;
+			else
+				name = replacing.substr(0, colon_pos);
+
+			static const auto stringicmp = [](const std::string_view& a, const std::string_view& b) {
```

Comment:
runtime.cpp is Windows only anyway, so simpler to just use `_stricmp` directly here (the `stringicmp` lambda is necessary in the effect compiler code only, since that has to support Linux as well, for other users like VkBasalt).